webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.js

```diff
@@ -0,0 +1,62 @@
+import React, { createContext } from 'react';
+import { Modal } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import ForemanModalHeader from './subcomponents/ForemanModalHeader';
+import ForemanModalFooter from './subcomponents/ForemanModalFooter';
+
+export const ModalContext = createContext(null);
+
+const ForemanModal = props => {
+  const { title } = props;
+  const { isOpen, onClose, ...propsToPassDown } = props; // don't pass down those two as it causes PF problems
```

Comment:
Wouldn't it pass the `children` prop as well?

Reply:
@sharvit good point!  I would think so, so I tested it but I'm not getting any children twice.  

When assembling props, I think React uses something like `Object.assign` to deal with it.  So, if you define a certain prop like this:
```
<MyComponent {...props} myProp=value />
```
the later definition overrides the earlier one.

So, it follows that if you do

```
const myProps = { children: 'blah' }
return (
  <MyComponent {...myProps}>
     foo
  </MyComponent>
);
```
the `children` prop that is defined later will override the earlier one, and your children will be `foo` and not `blah`.

So I think what's happening is, if you pass down children thru spread syntax and then also pass children in JSX, the JSX children prop overwrites the earlier children prop.